### PR TITLE
Adding the Index ID to explain_prediction_best_worst_output

### LINKED ISSUE #1119: Prediction explanations: display Index ID for best/worst prediction explanations
**Goal**
If users intend to understand the best and worst predictions, the API should allow them to see the Index of the explanations. 

If the user can provide the index column (perhaps through DataTables), that index ID could be displayed in the reported outputted by `explain_predictions_best_worst`. 

If the user doesn't provide an index column, no index ID should be displayed in the report.  

**Proposal**
Add index ID to the prediction explanation, if the user provides an index column in X.

Note the **Index ID:**
```
        Best 1 of 2

                Predicted Probabilities: [benign: 0.0, malignant: 1.0]
                Predicted Value: malignant
                Target Value: malignant
                Cross Entropy: 0.0
                Index ID: 45


                     Feature Name         Feature Value        Contribution to        SHAP Value
                                                                 Prediction
                ================================================================================
                    worst perimeter          155.30                   +                  0.10
                     worst radius             23.14                   +                  0.08
                 worst concave points         0.17                    +                  0.08
                worst fractal dimension       0.09                    -                 -0.00
                   compactness error          0.04                    -                 -0.00
                    worst symmetry            0.22                    -                 -0.00


        Best 2 of 2

                Predicted Probabilities: [benign: 0.0, malignant: 1.0]
                Predicted Value: malignant
                Target Value: malignant
                Cross Entropy: 0.0
                Index ID: 2

                    Feature Name       Feature Value   Contribution to Prediction   SHAP Value
                ==============================================================================
                  worst perimeter         166.10                   +                   0.10
                    worst radius           25.45                   +                   0.08
                worst concave points       0.22                    +                   0.08
                 compactness error         0.03                    -                  -0.00
                 worst compactness         0.21                    -                  -0.00
                   worst symmetry          0.21                    -                  -0.00


        Worst 1 of 2

                Predicted Probabilities: [benign: 0.552, malignant: 0.448]
                Predicted Value: benign
                Target Value: malignant
                Cross Entropy: 0.802
                Index ID: 7

                    Feature Name       Feature Value   Contribution to Prediction   SHAP Value
                ==============================================================================
                  smoothness error         0.00                    +                   0.04
                    mean texture           21.58                   +                   0.03
                   worst texture           30.25                   +                   0.02
                worst concave points       0.11                    -                  -0.02
                    worst radius           15.93                   -                  -0.03
                mean concave points        0.02                    -                  -0.03

```

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description

Fixes #1119.

FYI @RezaAkhtar

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Comment by RezaAkhtar** [comment]
LGTM 🆔

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 1]
```diff
@@ -345,6 +349,7 @@ def test_output_format_checked():
                 Predicted Value: setosa
                 Target Value: setosa
                 Cross Entropy: 0.15
+                Index ID: {{index_0}}
```
Comment (bchen1116) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:352]: why does this have two curly braces?

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 2]
```diff
@@ -426,12 +432,23 @@ def test_explain_predictions_best_worst_and_explain_predictions(mock_make_table,
     pipeline.problem_type = problem_type
     pipeline.name = "Test Pipeline Name"
 
+    def _add_custom_index(answer, index_best, index_worst, output_format):
```
Comment (bchen1116) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:435]: nice!

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py (hunk 1 above)
Comment (freddyaboulton) [reply evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:352]: We call `.format` once to format the string with the template for the multiclass shap table (that'll remove the outer curly braces) and then in the test we call `.format` to format the answer with the expected index (that'll remove the  inner curly braces).

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 3]
```diff
@@ -229,6 +229,7 @@ def test_output_format_checked():
                 Predicted Value: 1
                 Target Value: 2
                 Absolute Difference: 1
+                Index ID: {index_0}
```
Comment (gsheni) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:232]: Do you want to update the documentation to use this new feature?
- https://evalml.alteryx.com/en/latest/user_guide/model_understanding.html#Explaining-Multiple-Predictions

Comment (freddyaboulton) [reply evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:232]: @gsheni Can you verify that the index id shows up in the docs built for this branch?

https://evalml.alteryx.com/en/1119-index-id-best-worst/user_guide/model_understanding.html#Explaining-Multiple-Predictions

I'll also add a sentence to the documentation explaining what Index ID refers to!

Comment (gsheni) [reply evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:232]: Yep, they show up. Thanks!

**Review by angela97lin — APPROVED** [review]
Solid, LGTM! I see the updated docs as well. 🥳 
image

**Pull request merged**